src/classes/CRLP_Rollup_SEL_TEST.cls

```diff
@@ -48,59 +48,748 @@ public class CRLP_Rollup_SEL_TEST {
+                'There should be 2 rollups for the Account summary object');
+
+        System.assertNotEquals(null, CRLP_Rollup_SEL.getRollupIdByDeveloperName(devName),
+                'There should be one Rollup that matches that dev name');
+
+        System.assertNotEquals(null, CRLP_Rollup_SEL.getRollupById(mdtId),
+                'There should be one Rollup returned');
+
+        System.assertEquals(2, CRLP_Rollup_SEL.getRollupsById().size(),
+                'There should be 2 rollup definitions returned');
+
+        System.assertEquals(1, CRLP_Rollup_SEL.getRollupsByFilterGroup().size(),
+                'There should be 1 filter group that has rollups');
+    }
+
+    /**
+     * @description Tests CRLP_Rollup_SEL.UNSUPPORTED_OPERATIONS_BY_ROLLUP_FIELD is set as expected.
+     */
+    @IsTest
+    private static void shouldMatchUnsupportedOperationsMap() {
+
+        // This should be an exact duplicate of what is in CRLP_Rollup_SEL
+        Map<Schema.SObjectField, Set<String>> operationsMapForMatching = new Map<Schema.SObjectField, Set<String>> {
+                Schema.Rollup__mdt.Date_Field__c => new Set<String> {}, // All Operations Supported
+                Schema.Rollup__mdt.Amount_Field__c => new Set<String> {
+                        CRLP_Operation.RollupType.Count.name(),
+                        CRLP_Operation.RollupType.First.name(),
+                        CRLP_Operation.RollupType.Last.name()
+                }, // All Operations Supported
+                Schema.Rollup__mdt.Detail_Field__c => new Set<String> {
+                        CRLP_Operation.RollupType.Sum.name(),
+                        CRLP_Operation.RollupType.Count.name(),
+                        CRLP_Operation.RollupType.Average.name(),
+                        CRLP_Operation.RollupType.Years_Donated.name(),
+                        CRLP_Operation.RollupType.Donor_Streak.name(),
+                        CRLP_Operation.RollupType.Best_Year.name(),
+                        CRLP_Operation.RollupType.Best_Year_Total.name()
+                }
+        };
+
+        System.assertEquals(operationsMapForMatching, CRLP_Rollup_SEL.UNSUPPORTED_OPERATIONS_BY_ROLLUP_FIELD,
+            'CRLP_Rollup_SEL.UNSUPPORTED_OPERATIONS_BY_ROLLUP_FIELD has unexpected values'
+        );
+    }
+
+    /**
+     * @description Validate the validateReferencedFields() method to ensure it properly identifies bad fields
+     */
+    @IsTest
+    private static void shouldValidateFieldDescribes() {
+        Map<String, Schema.DescribeFieldResult> referencedFields = new Map<String, Schema.DescribeFieldResult>();
+        Map<String, Schema.DescribeFieldResult> allFields = UTIL_Describe.getAllFieldsDescribe('Account');
+
+        referencedFields.put('name', allFields.get('name'));
+        CRLP_Rollup_SEL.validateReferencedFields(referencedFields, 'Account');
+        System.assert(true, 'An exception was not thrown. Good!');
```

Comment:
This statement will not be reached if the previous command throws an exception.

Reply:
True, but a thrown exception would be bad and should stop the test. The Assert is there only to acknowledge that the call to validateReferencedFields() does not throw an exception.